vertical-pod-autoscaler/pkg/admission-controller/gencerts.sh

```diff
@@ -41,17 +41,18 @@ EOF
 # Create a certificate authority
 openssl genrsa -out ${TMP_DIR}/caKey.pem 2048
 set +o errexit
-openssl req -x509 -new -nodes -key ${TMP_DIR}/caKey.pem -days 100000 -out ${TMP_DIR}/caCert.pem -subj "/CN=${CN_BASE}_ca" -addext "subjectAltName = DNS:${CN_BASE}_ca"
-if [[ $? -ne 0 ]]; then
-  echo "ERROR: Failed to create CA certificate for self-signing. If the error is \"unknown option -addext\", update your openssl version or deploy VPA from the vpa-release-0.8 branch."
-  exit 1
-fi
+openssl req -x509 -new -nodes -key ${TMP_DIR}/caKey.pem -days 100000 -out ${TMP_DIR}/caCert.pem \
+    -subj "/CN=${CN_BASE}_ca" \
+    -config <(cat /etc/ssl/openssl.cnf \
```

Comment:
I don't quite understand this. It looks to me like we're passing output of printf to cat, and output of cat to stdin of openssl. Please explain

Reply:
Yup that's exactly what happens here. 
And finally aliases the generated file descriptor to openssl.cnf 

Example of `bash -x` output 
```
+ openssl req -x509 -new -nodes -key /tmp/caKey.pem -days 100000 -out /tmp/caCert.pem -subj /CN=segment_ca -config /dev/fd/63
++ cat /etc/ssl/openssl.cnf /dev/fd/63
+++ printf '\n[SAN]\nsubjectAltName=DNS:example_ca'
```

Reply:
Can you reorganize the code so that it is easier to read (use temp file or so)? Also, why do we need to modify openssl.cnf?